src/test/java/com/commercetools/sync/producttypes/utils/ProductTypesUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,129 @@
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.sphere.sdk.producttypes.ProductTypeDraft;
+import io.sphere.sdk.producttypes.commands.updateactions.ChangeDescription;
+import io.sphere.sdk.producttypes.commands.updateactions.ChangeName;
+import io.sphere.sdk.producttypes.commands.updateactions.SetKey;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+public class ProductTypesUpdateActionUtilsTest {
+    private static ProductType old;
+    private static ProductTypeDraft newSame;
+    private static ProductTypeDraft newDifferent;
+    private static ProductTypeDraft newWithNullValues;
+
+    /**
+     * Initialises test data.
+     */
+    @BeforeClass
+    public static void setup() {
+        final LocalizedString label = LocalizedString.of(Locale.ENGLISH, "label1");
+        final AttributeDefinition attributeDefinition = AttributeDefinitionBuilder
+                .of("attributeName1", label, StringAttributeType.of())
+                .build();
+
+        final List<AttributeDefinition> oldAttributeDefinitions = singletonList(attributeDefinition);
+
+        final List<AttributeDefinitionDraft> sameAttributeDefinitionDrafts = singletonList(
+                AttributeDefinitionDraftBuilder.of(attributeDefinition).build());
+
+        old = mock(ProductType.class);
+        when(old.getKey()).thenReturn("key1");
+        when(old.getName()).thenReturn("name1");
+        when(old.getDescription()).thenReturn("description 1");
+        when(old.getAttributes()).thenReturn(oldAttributeDefinitions);
+
+        newSame = ProductTypeDraft.ofAttributeDefinitionDrafts(
+                "key1",
+                "name1",
+                "description 1",
+                sameAttributeDefinitionDrafts
+        );
+
+        newDifferent = ProductTypeDraft.ofAttributeDefinitionDrafts(
+                "key2",
+                "name2",
+                "description 2",
+                sameAttributeDefinitionDrafts
+        );
+
+        newWithNullValues = ProductTypeDraft.ofAttributeDefinitionDrafts(
+                null,
+                "name3",
+                "description 3",
+                sameAttributeDefinitionDrafts
+        );
+    }
+
+    @Test
+    public void buildSetKeyAction_WithDifferentValues_ShouldReturnAction() {
+        final Optional<UpdateAction<ProductType>> result = buildSetKeyAction(old, newDifferent);
+        assertThat(result).isNotNull();
+        assertThat(result.isPresent()).isTrue();
+        assertThat(result.get()).isExactlyInstanceOf(SetKey.class);
+        assertThat(((SetKey) result.get()).getKey()).isEqualTo(newDifferent.getKey());
```

Comment:
instead of those 3 lines:
````java
assertThat(result).isNotNull();
assertThat(result.isPresent()).isTrue();
assertThat(result.get()).isExactlyInstanceOf(SetKey.class);
assertThat(((SetKey) result.get()).getKey()).isEqualTo(newDifferent.getKey());
````

you can also use the `AbstractOptionalAssert`  util from [`AssertJ` library](http://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/AbstractOptionalAssert.html) and do it in one line :)

````java
assertThat(result).containsInstanceOf(ChangeName.class);
assertThat(result).contains(ChangeName.of(newDifferent.getName()));
````

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/287/commits/fe004ec241788231f035ae6c439db07f7f322642